### PR TITLE
[Enhancement] add `SHOW PROC '/warehouses`

### PR DESCRIPTION
**Pull request by kevincai**
* add `SHOW PROC '/warehouses'`
* add into corresponding HTTP interface as well

Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 43 / 46 (93.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/warehouse/Warehouse.java|0|1|00.00%|[84]|
|:large_blue_circle:|com/starrocks/warehouse/WarehouseProcDir.java|26|28|92.86%|[44, 50]|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/LocalWarehouse.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowWarehousesStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/WarehouseClusterProcNode.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowWarehousesStmt.java [hunk 1]
```diff
@@ -26,6 +26,7 @@ public class ShowWarehousesStmt extends ShowStmt {
     private static final String WH_COL = "Warehouse";
     private static final ShowResultSetMetaData META_DATA =
             ShowResultSetMetaData.builder()
+                    .addColumn(new Column("Id", ScalarType.createVarchar(20)))
                     .addColumn(new Column(WH_COL, ScalarType.createVarchar(256)))
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowWarehousesStmt.java:30]:
```suggestion
                    .addColumn(new Column(Warehouse, ScalarType.createVarchar(256)))
```

File: test/sql/test_warehouse/R/test_warehouse [hunk 2]
```diff
@@ -0,0 +1,22 @@
+-- name: test_show_warehouse_clusters
```
Comment (wyb) [inline test/sql/test_warehouse/R/test_warehouse:1]: better to keep the same order as the test cases.

Comment (kevincai) [reply test/sql/test_warehouse/R/test_warehouse:10]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowWarehousesStmt.java (hunk 1 above)
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/ShowWarehousesStmt.java]: done

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 44 / 47 (93.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/warehouse/Warehouse.java|0|1|00.00%|[84]|
|:large_blue_circle:|com/starrocks/warehouse/WarehouseProcDir.java|26|28|92.86%|[44, 50]|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/LocalWarehouse.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowWarehousesStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/ProcService.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/WarehouseClusterProcNode.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|2|2|100.00%|[]|

**Review by abc982627271 — APPROVED** [review]

**Review by tracymacding — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**